.github/sccache/action.yml

```
@@ -0,0 +1,50 @@
+name: Install sccache
+description: Installs sccache for the current platform
+inputs:
+  read-only:
+    description: Whether the cache should be read-only
+    default: 'false'
+  cache-key:
+    description: The cache key to use to read/write the cache
+    default: ''
+    required: true
+runs:
+  using: 'composite'
+  steps:
+    - name: Set environment variables
+      run: echo "SCCACHE_VERSION=v0.3.3" >> $GITHUB_ENV
+      shell: bash
+    - name: Download sccache (Linux)
+      if: runner.os == 'Linux'
+      run: gh release download -R mozilla/sccache ${{ env.SCCACHE_VERSION }} -p sccache-${{ env.SCCACHE_VERSION }}-x86_64-unknown-linux-musl.tar.gz
+      shell: bash
+
+    - name: Download sccache (Windows)
+      if: runner.os == 'Windows'
+      run: gh release download -R mozilla/sccache ${{ env.SCCACHE_VERSION }} -p sccache-${{ env.SCCACHE_VERSION }}-x86_64-pc-windows-msvc.tar.gz
+      shell: bash
+
+    - name: Download sccache (MacOS)
+      if: runner.os == 'macOS'
+      run: gh release download -R mozilla/sccache ${{ env.SCCACHE_VERSION }} -p sccache-${{ env.SCCACHE_VERSION }}-x86_64-apple-darwin.tar.gz
+      shell: bash
```

What do you think about moving the if-else into bash, to reduce the amount of duplicated code?
```suggestion
    - name: Download sccache
      if: runner.os == 'Linux'
        run: |
          if [ $RUNNER_OS == "Linux"]; then
              PLATFORM="unknown-linux-musl"
          elif [ $RUNNER_OS == "Windows"]; then
              PLATFORM="pc-windows-msvc"
          elif [ $RUNNER_OS == "macOS"]; then
              PLATFORM="apple-darwin"
          fi
          gh release download -R mozilla/sccache ${{ env.SCCACHE_VERSION }} -p sccache-${{ env.SCCACHE_VERSION }}-x86_64-$PLATFORM.tar.gz
      shell: bash
```